File: 110-Balanced-Binary-Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private boolean check = true;
    public boolean isBalanced(TreeNode root) {
        heightCheck(root);
        return check;
    }
    public int heightCheck(TreeNode root){
        if(root == null) return 0;
        int left = heightCheck(root.left);
        int right = heightCheck(root.right);

        if(left > right && left - right > 1) check = false;
        else if(left < right && right -left > 1) check = false;
        
        return Math.max(left, right) +1;
    }
}

File: 110-Balanced-Binary-Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def sub_tree_height(node):
            if node is None:
                return 0
            left_tree = sub_tree_height(node.left)
            right_tree = sub_tree_height(node.right)
            if left_tree < 0 or right_tree < 0 or abs(left_tree-right_tree) > 1:
                return -1
            return 1 + max(left_tree , right_tree)
        return sub_tree_height(root) >= 0